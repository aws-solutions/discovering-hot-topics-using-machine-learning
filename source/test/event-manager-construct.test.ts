/**********************************************************************************************************************
* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
* SPDX-License-Identifier: Apache-2.0
 *********************************************************************************************************************/

import { Template } from 'aws-cdk-lib/assertions';
import { Aws, Stack } from 'aws-cdk-lib';
import { EventManager } from '../lib/integration/event-manager-construct';

test ('Event Bus creation', () => {
    const stack = new Stack();

    new EventManager(stack, 'TestEventManagerConstruct', {
        ruleConfig: [{
            eventPattern: {
                account: [ Aws.ACCOUNT_ID ],
                region: [ Aws.REGION ],
                resources: [ `arn:${Aws.PARTITION}:lambda:${Aws.REGION}:${Aws.ACCOUNT_ID}:function:*` ],
                source: [ 'com.platform.event.*' ]
            }
        }],
        eventBusName: 'TestEventBus'
    });
});