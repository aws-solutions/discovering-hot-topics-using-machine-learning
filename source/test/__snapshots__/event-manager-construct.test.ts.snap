// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Bus creation 1`] = `
{
  "Resources": {
    "EventRule07C93571D": {
      "Properties": {
        "EventBusName": {
          "Ref": "InferenceEventsC18A8A53",
        },
        "EventPattern": {
          "account": [
            {
              "Ref": "AWS::AccountId",
            },
          ],
          "region": [
            {
              "Ref": "AWS::Region",
            },
          ],
          "resources": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition",
                  },
                  ":lambda:",
                  {
                    "Ref": "AWS::Region",
                  },
                  ":",
                  {
                    "Ref": "AWS::AccountId",
                  },
                  ":function:*",
                ],
              ],
            },
          ],
          "source": [
            "com.platform.event.*",
          ],
        },
        "State": "ENABLED",
      },
      "Type": "AWS::Events::Rule",
    },
    "InferenceEventsC18A8A53": {
      "Properties": {
        "Name": "TestEventBus",
      },
      "Type": "AWS::Events::EventBus",
    },
  },
}
`;
